pages/unlock-publishing.tsx

```diff
@@ -0,0 +1,256 @@
+import { serverSideTranslations } from "next-i18next/serverSideTranslations";
+import { requireAuthentication } from "@lib/auth";
+import { checkPrivileges } from "@lib/privileges";
+
+import React, { useState } from "react";
+import { Formik } from "formik";
+import {
+  Button,
+  TextInput,
+  Label,
+  Alert,
+  ErrorListItem,
+  Description,
+  TextArea,
+} from "@components/forms";
+import { useTranslation } from "next-i18next";
+import * as Yup from "yup";
+import { isValidGovEmail } from "@lib/validation";
+import emailDomainList from "../email.domains.json";
+import { logMessage } from "@lib/logger";
+import { StyledLink } from "@components/globals/StyledLink/StyledLink";
+import axios from "axios";
+
+export default function UnlockPublishing() {
+  const { t, i18n } = useTranslation(["unlock-publishing", "common"]);
+  const [errorState, setErrorState] = useState({ message: "" });
+  const [submitting, setSubmitting] = useState(false);
+  const [hasRequestedPublishing, setHasRequestedPublishing] = useState(false);
+
+  const handleRequestPublishing = async (
+    managerEmail: string,
+    department: string,
+    goals: string
+  ) => {
+    return await axios({
+      url: "/api/request/publishXX",
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      data: { managerEmail, department, goals },
+      timeout: process.env.NODE_ENV === "production" ? 60000 : 0,
+    }).catch((err) => {
+      logMessage.error(err);
+      setErrorState({ message: t("submissionError") });
+    });
+  };
+
+  const validationSchema = Yup.object().shape({
+    department: Yup.string()
+      .required(t("input-validation.required", { ns: "common" }))
+      .max(50, t("signUpRegistration.fields.name.error.maxLength")),
+    managerEmail: Yup.string()
+      .required(t("input-validation.required", { ns: "common" }))
+      .email(t("input-validation.email", { ns: "common" }))
+      .test(
+        "managerEmail-govEmail",
+        t("input-validation.validGovEmail", { ns: "common" }),
+        (value = "") => isValidGovEmail(value, emailDomainList.domains)
+      ),
+    goals: Yup.string().required(t("input-validation.required", { ns: "common" })),
+  });
+
+  return (
+    <>
+      {!hasRequestedPublishing && (
+        <>
+          <Formik
+            initialValues={{ managerEmail: "", department: "", goals: "" }}
+            onSubmit={async ({ managerEmail, department, goals }) => {
+              setSubmitting(true);
+              try {
+                const response = await handleRequestPublishing(managerEmail, department, goals);
+                setSubmitting(false);
+                if (response.status !== 200) {
```

Comment:
Error handling needs more work in the future.

Reply:
I also tweaked `response.status` as `response` could be void if an error is thrown in `handleRequestPublising`.
For now just adding `response?.status` is good enough but you're right that the error handling could use a refactor in the future.